python/ray/serve/backend_worker.py

```diff
@@ -410,3 +411,27 @@ async def handle_request(self,
 
         self.num_ongoing_requests -= 1
         return result
+
+    async def shutdown(self):
+        """Perform graceful shutdown.
+
+        Trigger a graceful shutdown protocol that will wait for all the queued
+        tasks to be completed and then exit.
+        """
+        while True:
+            # Sleep first because we want to make sure all the routers receive
+            # the notification to remove this replica first.
+            await asyncio.sleep(DEFAULT_GRACEFUL_SHUTDOWN_LOOP_WAIT_S)
+
+            num_queries_waiting = self.batch_queue.qsize()
+            if (num_queries_waiting == 0) and (self.num_ongoing_requests == 0):
+                break
+            else:
+                logger.debug(
+                    f"Waiting for {DEFAULT_GRACEFUL_SHUTDOWN_LOOP_WAIT_S} "
+                    f"more to shutdown {self.replica_tag} because "
```

Comment:
```suggestion
                    f"Waiting for an additional {DEFAULT_GRACEFUL_SHUTDOWN_LOOP_WAIT_S}s "
                    f"to shut down replica {self.replica_tag} because "
```